fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionUtil.java

```diff
@@ -287,8 +289,9 @@ public static boolean isSyncWithPartition(MTMV mtmv, Long mtmvPartitionId,
      */
     public static String generatePartitionName(PartitionKeyDesc desc) {
         String partitionName = "p_";
-        partitionName += desc.toSql().trim().replaceAll("\\(|\\)|\\-|\\[|\\]|'|\\s+", "")
-                .replaceAll("\\(|\\)|\\,|\\[|\\]", "_");
+        Pattern pattern = Pattern.compile("[^a-zA-Z0-9,]");
+        Matcher matcher = pattern.matcher(desc.toSql());
+        partitionName += matcher.replaceAll("").replaceAll("\\,", "_");
```

Comment:
may generate same name for different partition

Reply:
Temporarily put on hold